functions-v2/src/on-analysis-document-imaged.ts

```diff
@@ -0,0 +1,95 @@
+import {FirestoreEvent, onDocumentCreated, QueryDocumentSnapshot} from "firebase-functions/v2/firestore";
+import * as logger from "firebase-functions/logger";
+import * as admin from "firebase-admin";
+import {getAnalysisQueueFirestorePath} from "./utils";
+import {categorizeUrl} from "../lib/src/ai-categorize-document";
+
+// This is one of three functions for AI analysis of documents:
+// 1. Watch for changes to the lastUpdatedAt metadata field and write a queue of docs to process
+// 2. Create screenshots of those documents
+// 3. (This function) Send those screenshots to the AI service for processing, and create comments with the results
+
+// NOTE: these should match the user specified in src/models/stores/user-types.ts
+const commenterName = "Ada Insight";
+const commenterUid = "ada_insight_1";
+
+const imagedQueuePath = getAnalysisQueueFirestorePath("imaged", "{docId}");
+
+async function error(error: string, event: FirestoreEvent<QueryDocumentSnapshot | undefined, Record<string, string>>) {
+  logger.warn("Error processing document", event.document, error);
+  const firestore = admin.firestore();
+  await firestore.doc(getAnalysisQueueFirestorePath("failedAnalyzing", event.params.docId)).set({
+    ...event.data?.data(),
+    error,
+  });
+  await firestore.doc(event.document).delete();
+}
+
+export const onAnalysisDocumentImaged =
+  onDocumentCreated(imagedQueuePath, async (event) => {
+    const {docId} = event.params;
+    const firestore = admin.firestore();
+    const queueDoc = event.data?.data();
+
+    if (queueDoc?.evaluator !== "categorize-design") {
+      await error(`Unexpected value for evaluator: ${queueDoc?.evaluator}`, event);
+      return;
+    }
+
+    const docImageUrl = event.data?.get("docImageUrl");
+
+    const completion = await categorizeUrl(docImageUrl);
+    const reply = completion?.choices[0].message;
+
+    if (reply?.refusal) {
+      await error(`AI refusal: ${reply.refusal}`, event);
+      return;
+    }
+    if (!reply?.parsed) {
+      await error("No response from AI", event);
+      return;
+    }
+    const tags = [reply.parsed.category];
+    const message = reply.parsed.discussion +
+      ` Key Indicators: ${reply.parsed.keyIndicators.join(", ")}`;
+
+    const commentsPath = `demo/AI/documents/${docId}/comments`;
```

Comment:
This simplified path to comment might not work in all cases. 

For non-networked teacher comments they are stored under `uid:[someId]_[docId]`. I can't remember if `someId` is the id of the teacher that is commenting or the student that owns the document. 

For networked teacher comments they are stored under `[networkName]_[docId]`.

The important thing to figure out is if the CLUE runtime will show both the comments stored under the `[docId]` and the comments under `[prefix]_[docId]`. If it doesn't show both the comments, then making do that is probably the right thing do. That seems safer than having this system figure out which prefix to store the comments under.